server/src/test/java/org/elasticsearch/index/mapper/TextFieldMapperTests.java

```diff
@@ -532,7 +532,12 @@ public void testFielddata() throws IOException {
             IllegalArgumentException.class,
             () -> disabledMapper.fieldType("field").fielddataBuilder(FieldDataContext.noRuntimeFields("test"))
         );
-        assertThat(e.getMessage(), containsString("Text fields are not optimised for operations that require per-document field data"));
+        assertThat(
+            e.getMessage(),
+            containsString(
+                "Field data is disabled on [field] in []. Text fields are not optimised for operations that require per-document field data"
```

Comment:
could you check why the index name is empty in the returned error? Is it a bug or is it due to the setup of the test?

Reply:
Because the empty string is specified to `fullyQualifiedIndexName` here (line 42).
https://github.com/elastic/elasticsearch/blob/43e2852016fa28d2892324291c4de14bfc62fa73/server/src/main/java/org/elasticsearch/index/fielddata/FieldDataContext.java#L33-L42
I think this happens only under unordinary situations like tests, as the use cases indicate.

Reply:
can you provide a different fielddatacontext instance here and specify the index so that we can test that we get it back as part of the error?

Reply:
I've pushed a change accordingly. Hope it meets the intention.

Reply:
yes, thanks!